### ISSUE TITLE
Combine precipitation amount and probability into one graph

### ISSUE THREAD

**Issue by Babber** [davidtakac/bura#68]
This is a polite request for improvement.

I think one needs to scroll unnecessarily to see the graph for Precipitation Totals. Plus I always find myself trying to unite the info presented in the graphs for Precipitation Totals (PT) and Chance of Precipitation (CoP) to figure out about the rain for a given day: *rain is likely at this hour, but only a small amount, then unlikely later, but it could rain more...* I mean these data make practical sense together anyways. It would be easier to interpret them on the same graph.

One way to do it is just printing the same bars and curve that we have now on the same graph with the y-axis labels of one of the graphs on the left and without filling the area under the curve for the CoP.

Another and maybe even better way to do it would be to leave out CoP altogether and the bars for the PT could have an opacity proportional to the CoP. Meaning that completely opaque bars would mean 100% chance, while completely transparent that is invisible bars would obviously mean dry weather. IMHO, this would make the interpretation of the data a lot more straightforward, smart and intuitive.

Thank you for your consideration!

**Comment by devguy23857848467** [comment]
+1 to this great idea, combine these graphs would be less scrolling, and perhaps negate the need to have swiping between days lock to the same scroll position.

**Comment by davidtakac** [comment]
Great idea. I'll see what I can do!

**Comment by davidtakac** [comment]
Checked this out now. An issue is that sometimes Open-Meteo reports a non-zero probability of precipitation, but zero precipitation amount. See the screenshot. It would be confusing to have 20% precipitation in an hour, but no rain/snow/showers at the same time. 

![Image](https://github.com/user-attachments/assets/17c805aa-7511-4325-b486-ee601d656225)

**Comment by davidtakac** [comment]
But maybe it wouldn't be *that* confusing. Low probabilities with no rain amount could just mean "it might drizzle a bit". High probabilities will most likely have an amount associated as well. What do you guys think?

**Comment by Babber** [comment]
Yes, I have also noticed that sometimes there's a chance for some 0 mm rain and I always found it quite confusing rather than informative. What kind of probability distribution for precipitation gives you a more than zero chance for getting wet at all but less than 0.05 mm(?) expectation value for the total? Is this a precision issue? Seems somewhat inconsistent or at least contraintuitive to me. I would just ignore these low probabilities with no rain amount, like you last suggested @davidtakac.

**Comment by davidtakac** [comment]
> Yes, I have also noticed that sometimes there's a chance for some 0 mm rain and I always found it quite confusing rather than informative.

Same here. This might have something to do with Open-Meteo's implementation. It's an API that aggregates multiple weather sources for a given location, and two independent sources might not agree when it comes to precipitation amount and probability. 

>  I would just ignore these low probabilities with no rain amount

Would it make sense to do this on the app level? For example, if probability is less than 20% and precipitation amount is 0, just override the API and say there will be no rain for that period. Or do you think slim chances like that are also valuable?

**Comment by Babber** [comment]
I have just found this: [precipitation_probability > 0 but precipitation = 0](https://github.com/open-meteo/open-meteo/issues/783). So to my best understanding, we are essentially dealing here with two related but still different weather forecasts, which can of course contradict to some extent.

Maybe there is a way to keep both: just below the time axis, we could have a thick grey-scale line where black (or rather dark blue?) means 100% chance for precipitation and white 0% (and maybe the other way around in a dark theme). And then independently of this, above the time axis, we could just print vertical bars corresponding to the predicted amount of the rain.

What do you think?

**Comment by davidtakac** [comment]
Similarly to your suggestion, I was thinking of making the precipitation amount bars transparent according to the probability. 100% is completely opaque, 50% is half transparent, etc. However, differences in transparency are not that easily recognizable, so it would be a bit unclear. That can be addressed easily by just printing the probability percentage above/below the bar. I'll try a couple things out when I start working on this and post screenshots here so we can figure out what might work best.
